bin/env.sh

```diff
@@ -16,77 +16,89 @@ drop_from_path()
+      drop_from_path "$PATH" ${old_projectbase}/external/miniconda/bin
       PATH=$newpath
    fi
    if [ -n "${PYTHONPATH}" ]; then
-      drop_from_path $PYTHONPATH ${old_ntpbase}/python
+      drop_from_path $PYTHONPATH ${old_projectbase}/python
       PYTHONPATH=$newpath
    fi
 fi
 
-
 if [ -z "${PATH}" ]; then
-   PATH=$NTPROOT/bin; export PATH
+   PATH=$HEP_PROJECT_ROOT/bin; export PATH
 else
-   PATH=$NTPROOT/bin:$PATH; export PATH
+   PATH=$HEP_PROJECT_ROOT/bin:$PATH; export PATH
 fi
 
 if [ -z "${PYTHONPATH}" ]; then
-   PYTHONPATH=$NTPROOT/python; export PYTHONPATH
+   PYTHONPATH=$HEP_PROJECT_ROOT/python; export PYTHONPATH
 else
-   PYTHONPATH=$NTPROOT/python:$PYTHONPATH; export PYTHONPATH
+   PYTHONPATH=$HEP_PROJECT_ROOT/python:$PYTHONPATH; export PYTHONPATH
 fi
 
-# for external dependencies
-PYTHONUSERBASE=$NTPROOT/external; export PYTHONUSERBASE
-NEWPP=$(echo "$PYTHONUSERBASE"/lib/python*/site-packages)
-NEWPP=$(echo ${NEWPP// /:})
-PYTHONPATH=$NEWPP:$PYTHONPATH; export PYTHONPATH
-PATH=$PYTHONUSERBASE/bin:$PATH; export PATH
-PATH=/software/miniconda/bin:$PATH; export PATH
+unset old_projectbase
+unset envscript
 
 # for CMSSW
-source /cvmfs/cms.cern.ch/cmsset_default.sh
-export CMSSW_GIT_REFERENCE=/cvmfs/cms.cern.ch/cmssw.git
+if [ -f /cvmfs/cms.cern.ch/cmsset_default.sh ]; then
+	source /cvmfs/cms.cern.ch/cmsset_default.sh
+	export CMSSW_GIT_REFERENCE=/cvmfs/cms.cern.ch/cmssw.git
+fi
 
 # CRAB submission
 # https://twiki.cern.ch/twiki/bin/view/CMSPublic/CRAB3Releases#Improvements_enhancements_change
-#source /cvmfs/cms.cern.ch/crab3/crab_light.sh #funky behaviour if not used within CMSSW
-source /cvmfs/cms.cern.ch/crab3/crab.sh
+if [ -f /cvmfs/cms.cern.ch/crab3/crab.sh ]; then
+	source /cvmfs/cms.cern.ch/crab3/crab.sh
+fi
 
 # for grid tools
-#source /cvmfs/grid.cern.ch/etc/profile.d/setup-cvmfs-ui.sh
+vomsInfo=`which voms-proxy-info &> /dev/null`
+if [ "$vomsInfo" = "" ]; then
+  if [ -f /cvmfs/grid.cern.ch/etc/profile.d/setup-cvmfs-ui.sh ]; then
+	source /cvmfs/grid.cern.ch/etc/profile.d/setup-cvmfs-ui.sh
+  else
+    echo "Cannot find voms-proxy-info nor setup-cvmfs-ui.sh"
+  fi
+fi
 
-# for hadoop, needs to run after grid tools
-#if [[ -d "/usr/java/jdk1.7.0_67-cloudera" ]]; then
-#	export JAVA_HOME=/usr/java/jdk1.7.0_67-cloudera
-#	export PATH=$JAVA_HOME/bin:$PATH
-#else
-#	# use system default
-#	export JAVA_HOME=
-#fi
+# miniconda setup for modern python and additional python packages
+if [ ! -d "${HEP_PROJECT_ROOT}/external" ] ; then
+	mkdir ${HEP_PROJECT_ROOT}/external
+fi
 
-unset old_ntpbase
-unset envscript
+if [ ! -d "${HEP_PROJECT_ROOT}/external/miniconda" ] ; then
+	wget -nv http://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O miniconda.sh
+	bash miniconda.sh -b -p ${HEP_PROJECT_ROOT}/external/miniconda
+	PATH=${HEP_PROJECT_ROOT}/external/miniconda/bin:$PATH; export PATH
+	rm -f miniconda.sh
+	conda update conda -y
+	conda update pip -y
+	conda install git wget -y
```

Comment:
The git install turns out to be 210 MB inside the tarball. The reason for that are the symlinks.
Instead of retaining symlinks, tarring, or to be more precise ` shutil.make_archive`, will create duplicates of the file the symlink is pointing to.
two options:
 - [ ] either skip the git install
 - [ ] or fix `shutil.make_archive`